src/librustc_typeck/diagnostics.rs

```diff
@@ -3426,6 +3426,37 @@ parameters. You can read more about it in the API documentation:
 https://doc.rust-lang.org/std/marker/struct.PhantomData.html
 "##,
 
+E0393: r##"
+A type parameter which references `Self` in its default value was not specified.
+Example of erroneous code:
+
+```compile_fail
+trait A<T=Self> {}
+
+fn together_we_will_rule_the_galaxy(son: &A) {}
+// error: the type parameter `T` must be explicitly specified in an
+          object type because its default value `Self` references the
```

Comment:
```Shell
<stdin>:5:49: 5:50 error: unknown start of token: `
<stdin>:5               object type because its default value `Self` references the
```

It seems it doesn't like backticks in here. Should I remove them?